src/Tasks/Microsoft.NET.Build.Tasks/ResolvePublishAssemblies.cs

```diff
@@ -53,13 +54,14 @@ protected override void ExecuteCore()
             IEnumerable<ResolvedFile> resolvedAssemblies = 
                 new PublishAssembliesResolver(packageResolver)
                     .WithPrivateAssets(privateAssetsPackageIds)
+                    .PreserveCacheLayout(PreserveCacheLayout)
```

Comment:
This method should follow the same naming as the existing method.  `WithPreserveCacheLayout`.